Gems/Atom/Feature/Common/Code/Source/CoreLights/DirectionalLightFeatureProcessor.cpp

```diff
@@ -437,6 +436,22 @@ namespace AZ
 
         // --- Cascade Shadows ---
 
+        void DirectionalLightFeatureProcessor::SetShadowEnabled(LightHandle handle, bool enable)
+        {
+            if (enable)
```

Comment:
I might missed that, but what is the logic to disable shadows if none is enabled?

Reply:
There is already some mechanism to handle the case of disabling shadow. When setting the configurations to the shadow pass, the FP would see whether the current light handle equals to `m_shadowingLightHandle`. So if the `m_shadowingLightHandle` not equal to the current handle, those settings would be returned directly.

If all dir lights are turned off shadow, the `m_shadowingLightHandle` is initialized as `Null`, not equals to any of the existing light handle, then no dir light shadow is casted.

You can search for the `m_shadowingLightHandle` in `DirectionalLightFeatureProcessor.cpp`.